include/circt/Dialect/FIRRTL/FIRRTLTypes.td

```diff
@@ -45,73 +45,73 @@ def ForeignType : FIRRTLDialectType<CPred<"!$_self.isa<FIRRTLType>()">,
  "OpenBundleType", "::circt::firrtl::OpenBundleType">;
 
-def FVectorType : FIRRTLDialectType<CPred<"$_self.isa<FVectorType>()">,
-  "FVectorType", "::circt::firrtl::FVectorType">;
-def OpenVectorType : FIRRTLDialectType<CPred<"$_self.isa<OpenVectorType>()">,
+def FVectorType : FIRRTLDialectType<CPred<"firrtl::type_isa<FVectorType>($_self)">,
+  "FVectorType", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::FVectorType>">;
+def OpenVectorType : FIRRTLDialectType<CPred<"firrtl::type_isa<OpenVectorType>($_self)">,
   "OpenVectorType", "::circt::firrtl::OpenVectorType">;
 
-def FEnumType : FIRRTLDialectType<CPred<"$_self.isa<FEnumType>()">,
- "FEnumType", "::circt::firrtl::FEnumType">;
+def FEnumType : FIRRTLDialectType<CPred<"firrtl::type_isa<FEnumType>($_self)">,
+ "FEnumType", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::FEnumType>">;
 
 def StringType : FIRRTLDialectType<CPred<"$_self.isa<StringType>()">,
   "StringType", "::circt::firrtl::StringType">;
 
 def AggregateType : FIRRTLDialectType<
   Or<[
-    CPred<"$_self.isa<FVectorType>()">,
-    CPred<"$_self.isa<BundleType>()">,
-    CPred<"$_self.isa<FEnumType>()">
+    CPred<"firrtl::type_isa<FVectorType>($_self)">,
+    CPred<"firrtl::type_isa<BundleType>($_self)">,
+    CPred<"firrtl::type_isa<FEnumType>($_self)">
   ]>,
   "a aggregate type", "::circt::firrtl::FIRRTLBaseType">;
 
-def SizedType : FIRRTLDialectType<CPred<"$_self.isa<FIRRTLBaseType>() && "
-    "!$_self.cast<FIRRTLBaseType>().hasUninferredWidth()">,
+def SizedType : FIRRTLDialectType<CPred<"isa<FIRRTLBaseType>($_self) && "
+    "!cast<FIRRTLBaseType>($_self).hasUninferredWidth()">,
     "a sized type (contains no uninferred widths)", "::circt::firrtl::FIRRTLType">;
 def SizedOrForeignType : AnyTypeOf<[SizedType, ForeignType]>;
 
 def AsyncResetType : FIRRTLDialectType<
-    CPred<"$_self.isa<AsyncResetType>()">,
-    "AsyncReset", "::circt::firrtl::AsyncResetType">;
+    CPred<"firrtl::type_isa<AsyncResetType>($_self)">,
+    "AsyncReset", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::AsyncResetType>">;
 
 def ResetType : FIRRTLDialectType<
-    CPred<"$_self.isa<ResetType>()">,
-    "Reset", "::circt::firrtl::ResetType">;
+    CPred<"firrtl::type_isa<ResetType>($_self)">,
+    "Reset", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::ResetType>">;
 
 def PassiveType : FIRRTLDialectType<
-  CPred<"$_self.isa<FIRRTLBaseType>() && $_self.cast<FIRRTLBaseType>().isPassive()">,
+  CPred<"isa<FIRRTLBaseType>($_self) && cast<FIRRTLBaseType>($_self).isPassive()">,
   "a passive base type (contain no flips)", "::circt::firrtl::FIRRTLBaseType">;
 
 def RefType : FIRRTLDialectType<
-  CPred<"$_self.isa<RefType>()">,
+  CPred<"firrtl::type_isa<RefType>($_self)">,
    "reference type", "::circt::firrtl::RefType">;
 
 def RWProbe : FIRRTLDialectType<
-  CPred<"$_self.isa<RefType>() && $_self.cast<RefType>().getForceable()">,
+  CPred<"firrtl::type_isa<RefType>($_self) && cast<RefType>($_self).getForceable()">,
```

Comment:
Don't mix firrtl::type_* and llvm's casting methods, right?

Anyway this has similar problem as above re:allowing aliases but then casting as if not an alias (= assertion failure).  Either it's wrong or unreachable, I think?